src/components/editor/EditorLogin.jsx

```diff
@@ -1,74 +1,92 @@
 import BaseComponent from 'lib/BaseComponent';
 import React from 'react';
 import { browserHistory } from 'react-router';
-import { hash } from 'lib/utilities';
+import { googleClientID } from 'lib/utilities';
 import _ from 'lodash';
 
-import TextField from 'material-ui/TextField';
-import RaisedButton from 'material-ui/RaisedButton';
-
-const styles = {
-  buttons: {
-    marginTop: 12,
-    marginBottom: 24,
+const style = {
+  button: {
+    marginTop: 36,
   },
 };
 
 export default class Login extends BaseComponent {
   constructor(props) {
     super(props);
-    this.safeSetState({
-      passwordValue: '',
-    });
-    this.handlePasswordChange = this.handlePasswordChange.bind(this);
-    this.handleSubmit = this.handleSubmit.bind(this);
+    this.state = {
+      gapiReady: false,
+    };
+    this.handleRedirect = this.handleRedirect.bind(this);
+    this.onSignIn = this.onSignIn.bind(this);
+    this.onSignInSuccess = this.onSignInSuccess.bind(this);
   }
 
-  handlePasswordChange(e) {
-    this.safeSetState({
-      passwordValue: e.target.value,
-    });
-  }
+  componentDidMount() {
```

Comment:
I generally love this new implementation, but the only thing I don't like about this is that this will run every time you go back to the Login component, so for example if you sign out right? Then the script would be loaded again.

So I think I see two ways of doing this, 1 is doing this in `componentDidMount` in the `EditorAppController`, but I think even easier is just adding it to the initial HTML file we send and add it to the `<head>` tag there, which you can find at the top of the `admin-server.js` file. Sounds good? You should still be able to do something asynchronous about waiting for the JS file to load right?

Reply:
okay I'll test if this works

Reply:
I'm actually not sure how to do the script`onload` in React, mind give a hint? 
I tried moving it to `EditorAppController`, but then the page redirect breaks for some reason.

Reply:
What you would probably do is move the `gapiReady` state to `EditorAppController` and then pass it as a prop to `EditorLogin`, that addresses the problem you're talking about right?

Also, you could probably rename `gapiReady` to something more comprehensible such as `GoogleAPIReady` or something like that

Reply:
Another option is also to keep it as it is, but do a check possibly with `document.querySelectorAll` to check all the scripts and see if one of them has the `src` tag already (aka you already loaded it once). 

You could possibly also just put it in the initial HTML as I talked about and then use something such as the first answere [here](https://stackoverflow.com/questions/538745/how-to-tell-if-a-script-tag-failed-to-load) with a `setInterval` that you clear when it finds the variable or something like thatAnother option is also to keep it as it is, but do a check possibly with `document.querySelectorAll` to check all the scripts and see if one of them has the `src` tag already (aka you already loaded it once). 

You could possibly also just put it in the initial HTML as I talked about and then use something such as the first answere [here](https://stackoverflow.com/questions/538745/how-to-tell-if-a-script-tag-failed-to-load) with a `setInterval` that you clear when it finds the variable or something like that

Reply:
I was just going to say, the first way is probably much easier to implement. If I'm not mistaken, `EditorLogin` is not a children of `EditorAppController`, so passing props can be difficult (?). I'll check out the links. many thanks!

Reply:
Yeah, you're right, you would need some tricks, or some global state management like Redux, so for now it's probably easier going with the above. 

Also just saw it looks like I accidentally did some weird copy pasting in the above message and wrote the same message twice in a row haha, I'll edit that now